lib/pure/ioselects/ioselectors_kqueue.nim

```diff
@@ -584,16 +584,19 @@ proc select*[T](s: Selector[T], timeout: int): seq[ReadyKey] =
 template isEmpty*[T](s: Selector[T]): bool =
   (s.count == 0)
 
-proc getData*[T](s: Selector[T], fd: SocketHandle|int): T =
+proc contains*[T](s: Selector[T], fd: SocketHandle|int): bool {.inline.} =
+  return s.fds[fd].ident != 0
+
+proc getData*[T](s: Selector[T], fd: SocketHandle|int): var T =
   let fdi = int(fd)
   s.checkFd(fdi)
-  if s.fds[fdi].ident != 0:
+  if fdi in s:
```

Comment:
This is not a correct replacement. 'fdi' is always in s.

Reply:
Huh? I must be missing something.

``fdi in s`` -> ``contains(s, fdi)`` -> ``s.fds[fdi].ident != 0`` (equivalent to what it was)

Reply:
ok